src/engine-actions/get-status.js

```diff
@@ -0,0 +1,77 @@
+const grpc = require('grpc')
+
+const {
+  getInfo,
+  genSeed
+} = require('../lnd-actions')
+
+/**
+ * @constant
+ * @type {String}
+ * @default
+ */
+const WALLET_EXISTS_ERROR_MESSAGE = 'wallet already exists'
+
+/**
+ * Returns the state of the current lnd-engine.
+ * @see {LndEngine#ENGINE_STATUS}
+ * @returns {String} status - ENGINE_STATUS
+ */
+function getStatus () {
+  return new Promise((resolve, reject) => {
```

Comment:
why implement this as a Promise instead of `async`? And same question for each subsequent call - I think `await` with a try/catch wrapper is more readable than this

Reply:
@treygriffith multiple try catches were super ugly before, but ill try it again and see if it clears anything up